flink/json-taskmanagers.yml

```diff
@@ -0,0 +1,120 @@
+type: json
+data:
+  url: /v1/taskmanagers
+  server: ${FLINK_HOSTPORT}
+
+observation:
+  - name: taskmanagers
+    metricNamespace: flink
+    path: $.taskmanagers[?(@.id=${taskmanagerid})]
+
+    metric:
+      - name: taskmanagers.cores
+        source: json:/v1/taskmanagers/${taskmanagerid} $.hardware.cpuCores
+        type: long_gauge
+        label: cpu cores
+        description: cpu cores
+
+      - name: taskmanagers.memory.total
+        source: json:/v1/taskmanagers/${taskmanagerid} $.hardware.physicalMemory
+        type: long_gauge
+        unit: bytes
+        label: total memory
+        description: physical memory
+
+      - name: taskmanagers.memory.managed
+        source: json:/v1/taskmanagers/${taskmanagerid} $.hardware.managedMemory
+        type: long_gauge
+        unit: bytes
+        label: managed memory
+        description: managed memory
+
+      - name: taskmanagers.heap.used
+        source: json:/v1/taskmanagers/${taskmanagerid} $.metrics.heapUsed
+        type: long_gauge
+        unit: bytes
+        label: used heap
+        description: The amount of heap memory currently used
+
+      - name: taskmanagers.heap.committed
+        source: json:/v1/taskmanagers/${taskmanagerid} $.metrics.heapCommitted
+        type: long_gauge
+        unit: bytes
+        label: committed heap
+        description: The amount of heap memory guaranteed to be available to the JVM
+
+      - name: taskmanagers.heap.max
+        source: json:/v1/taskmanagers/${taskmanagerid} $.metrics.heapMax
+        type: long_gauge
+        label: max heap
+        description: The maximum amount of heap memory that can be used for memory management
+        unit: bytes
+
+      - name: taskmanagers.nonheap.used
+        source: json:/v1/taskmanagers/${taskmanagerid} $.metrics.nonHeapUsed
+        type: long_gauge
+        unit: bytes
+        label: used non-heap
+        description: The amount of non-heap memory currently used
+
+      - name: taskmanagers.nonheap.committed
+        source: json:/v1/taskmanagers/${taskmanagerid} $.metrics.nonHeapCommitted
+        type: long_gauge
+        unit: bytes
+        label: committed non-heap
+        description: The amount of non-heap memory guaranteed to be available to the JVM
+
+      - name: taskmanagers.direct.count
```

Comment:
Same naming comment as for related jobmanagers metric.

Reply:
Fixed